tools/deploy_wiremock.py

```diff
@@ -23,6 +34,86 @@ def deploy_wiremock(deploy_options):
         file=dst_file
     )
 
+    # Ensure the Wiremock pod is ready before populating the stubs.
+    # To verify if Wiremock is ready to receive requests,
+    # we need to obtain its hostname and port.
+    log.info("Waiting for Wiremock service to be ready...")
+    wait_for_wiremock_service(namespace=deploy_options.namespace)
+    log.info("Wiremock service is ready")
+    hostname = get_service_external_ip(namespace=deploy_options.namespace)
+    port = get_service_external_port(namespace=deploy_options.namespace)
+
+    url = f"http://{hostname}:{port}/__admin/mappings"
+    log.info(f"Trying to reach Wiremock is on {url}...")
+    wait_for_wiremock_pod(url=url)
+    log.info(f"Wiremock is accessible")
+    populate_stubs(hostname=hostname, port=port)
+    log.info("Wiremock stubs populated")
+
+def get_service_external_ip(namespace: str) -> str:
+    return utils.check_output(f"kubectl get service wiremock -n {namespace} -ojson | jq -r '.status.loadBalancer.ingress[0].ip // .status.loadBalancer.ingress[0].hostname'")
+
+
+def get_service_external_port(namespace: str) -> str:
+    return utils.check_output(f"kubectl get service wiremock -n {namespace} -ojson | jq -r '.spec.ports[0].port'")
+
+
+def deploy_ingress(hostname, deploy_options):
+    src_file = os.path.join(os.getcwd(), 'deploy', "wiremock", "wiremock-ingress.yaml")
+    dst_file = os.path.join(os.getcwd(), 'build', deploy_options.namespace, "wiremock-ingress.yaml")
+    with open(src_file, "r") as src:
+        with open(dst_file, "w+") as dst:
+            data = src.read()
+            data = data.replace('REPLACE_NAMESPACE', deploy_options.namespace)
+            data = data.replace("REPLACE_HOSTNAME", hostname)
+            log.info(f"Deploying {dst_file}")
+            dst.write(data)
```

Comment:
can we use here `yaml.load` and update the value as a dict.
and `yaml.dump` to write it to a file.

Reply:
I agree, I was just following the pattern in other scripts for consistency, also this is how the code in master already is